src/linear_algebra/tensor_product.lean

```diff
@@ -632,6 +634,9 @@ def curry (f : M ⊗ N →ₗ P) : M →ₗ N →ₗ P := lcurry R M N P f
 @[simp] theorem curry_apply (f : M ⊗ N →ₗ[R] P) (m : M) (n : N) :
   curry f m n = f (m ⊗ₜ n) := rfl
 
+lemma curry_inj : function.injective (curry : (M ⊗[R] N →ₗ[R] P) → (M →ₗ[R] N →ₗ[R] P)) :=
+λ g h H, by { ext m n, simp only [linear_map.compr₂_apply, mk_apply, ← curry_apply, H], }
```

Comment:
I think this statement is defeq to `mk_compr₂_inj`? So you can reuse the proof.

Reply:
Good call!